sql/wsrep_thd.cc

```diff
@@ -676,3 +680,36 @@ bool wsrep_thd_has_explicit_locks(THD *thd)
   assert(thd);
   return thd->mdl_context.has_explicit_locks();
 }
+
+/*
+  Get auto increment variables for THD. Use global settings for
+  applier threads.
+ */
+void wsrep_thd_auto_increment_variables(THD* thd,
+                                        unsigned long long* offset,
+                                        unsigned long long* increment)
+{
+  if (thd->wsrep_exec_mode == REPL_RECV &&
+      thd->wsrep_conflict_state != REPLAYING)
+  {
+    *offset= global_system_variables.auto_increment_offset;
+    *increment= global_system_variables.auto_increment_increment;
+  }
+  else
+  {
+    *offset= thd->variables.auto_increment_offset;
+    *increment= thd->variables.auto_increment_increment;
+  }
+}
+
+bool wsrep_safe_to_persist_xid(THD* thd)
+{
+  /* Rollback of transaction too also invoke persist of XID.
+  Avoid persisting XID in this use-case. */
+  bool safe_to_persist_xid= false;
+  if (thd->wsrep_conflict_state == NO_CONFLICT      ||
+      thd->wsrep_conflict_state == REPLAYING        ||
+      thd->wsrep_conflict_state == RETRY_AUTOCOMMIT)
+    safe_to_persist_xid= true;
+  return(safe_to_persist_xid);
+}
```

Comment:
Something strange about the function: in the original PXC the function is used by `trx_write_serialisation_history()`, but you do not use it at all - there are only definitions and declarations of the function, but no one use.